app/views/bgeigie_imports/index.html.erb

```diff
@@ -51,6 +51,17 @@
 <%- if @bgeigie_imports.empty? -%>
   <%= t("none_found") -%>
 <%- else -%>
+
+  <ul class="inline pull-right">
+    <%-if user_signed_in? %>
+      <%- if current_user.moderator? -%>
+        <li id="open_all" data-bgeigie="<%=@bgeigie_imports.to_json%>">
```

Comment:
Seems like it'd be better to have this be an id list, or possibly better yet get the list of IDs out of the table. My guess is this leads to a very very large json blob in the middle of the HTML.

Reply:
But if I get the list of IDs out of the table, the button has to be under the tabe right? It would be smoother for the moderation flow if the button would be over the table.

Reply:
No you can have the ids in the erb (might be there already) then have the coffescript/javascript read them out. The javascript won't execute until clicked so page location won't matter

Reply:
Since we have `@bgeigie_imports` here, we can get list of IDs like `@bgeigie_imports.map(&:id)` (or `@bgeigie_imports.map { |i| i.id }`.  If we want to filter with unapproved imports, then `@bgeigie_imports.each_with_object([]) { |import, ary| ary << import.id unless import.approved? }`. I think that filtering should be done in Rails application side because application knows what is approved import.

Currently, URLs for import pages are generated in `open_all.js.coffee`, but it would be better to generate URLs in views or controllers and pass these URLs to JavaScript. So, it would be like:

```erb
<%= link_to t('.Open all'), '#', class: 'btn', id: 'open_all', data: { urls: @bgeigie_imports.each_with_object([]) { |import, ary| ary << bgeigie_imports_path(i) unless import.approved? }.to_json } %>
```

And, in JavaScript, you just iterate list of URLs that are passed via `data-urls` attribute , and open each.

Reply:
Let's please use a view helper though and not pack that much Ruby into the erb 😉

Reply:
But I agree with eito's point. The table doesn't know about moderation status so assembling a list in erb seems like probably a good move. We may want to relabel the button though. "Open all unmoderated" or something.